tcpproxy/tcpproxy.go

```diff
@@ -247,19 +262,24 @@ func (t *testTCPProxy) copy(
 					string(buf[:nr]))
 			}
 
+			t.mu.Unlock()
```

Comment:
I think you meant to use `.Lock()` here.

It'll be better to wrap the whole thing in 
```
func() {
  ...lock and defer unlock...
  ...
}()
```
?